test/e2e/storage_test.go

```diff
@@ -40,7 +44,14 @@ func testStorage(t *testing.T) {
 		pvc, err = vsphereProvider.CreateInTreePVC(tc.client.K8s, tc.workloadNamespace)
 		require.NoError(t, err)
 	} else {
-		pvc, err = tc.CloudProvider.CreatePVC(tc.client.K8s, tc.workloadNamespace)
+		var pv *core.PersistentVolume
+		if tc.CloudProvider.GetType() == config.NonePlatformType {
+			err = smb.EnsureSMBControllerResources(tc.client.K8s)
+			require.NoError(t, err)
+			pv, err = tc.createSMBPV()
+			require.NoError(t, err)
+		}
+		pvc, err = tc.CloudProvider.CreatePVC(tc.client.K8s, tc.workloadNamespace, pv)
```

Comment:
`pv` cannot be nil, so consider moving this inside the if block after creating it.

Reply:
Doesn't this need to work for both the platform none and non-platform none case? If yes, this cannot move into the if block.

Reply:
Correct it cannot be moved